Makefile

```diff
@@ -327,7 +327,7 @@ golangci-lint: ## Download golangci-lint locally if necessary.
 CONTROLLER_GEN = $(PROJECT_DIR)/bin/controller-gen
 .PHONY: controller-gen
 controller-gen: ## Download controller-gen locally if necessary.
-	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install sigs.k8s.io/controller-tools/cmd/controller-gen
+	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
```

Comment:
`controller-gen` should use the version set in go.mod

Reply:
Indeed. Please revert this line, as the version is locked in go.mod  https://github.com/kubernetes-sigs/kueue/pull/1720

Reply:
> Indeed. Please revert this line, as the version is locked in go.mod #1720
I am a novice in Go and Kubernetes development. I wanted to address the following issue, which is why I changed that line of code.
<img width="1382" alt="image" src="https://github.com/kubernetes-sigs/kueue/assets/11625846/ea6e3485-8d2b-4c8c-960a-94dfe1876607">